flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTableUtil.java

```diff
@@ -138,4 +148,88 @@ public static Partition createHivePartition(String dbName, String tableName, Lis
 		return partition;
 	}
 
+	/**
+	 * Generates a filter string for partition columns from the given filter expressions.
+	 *
+	 * @param partColNames The names of all partition columns
+	 * @param expressions  The filter expressions in CNF form
+	 * @return a filter string equivalent to the expressions, or null if the expressions can't be handled
+	 */
+	public static String makePartitionFilter(List<String> partColNames, List<Expression> expressions) {
+		List<String> filters = new ArrayList<>(expressions.size());
+		ExpressionExtractor extractor = new ExpressionExtractor(partColNames);
+		for (Expression expression : expressions) {
+			String str = expression.accept(extractor);
+			if (str == null) {
```

Comment:
return Optional?

Reply:
I changed the public method to return an Optional.